src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/TextCompletionItemProvider.cs

```diff
@@ -0,0 +1,96 @@
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+
+namespace Microsoft.CodeAnalysis.Razor.Completion
+{
+    internal class TextCompletionItemProvider : RazorCompletionItemProvider
+    {
+        private static readonly string TextTag = "text";
+        private static readonly HashSet<string> ElementCommitCharacters = new HashSet<string>{" ", ">"};
+        private readonly HtmlFactsService _htmlFactsService;
+
+        public TextCompletionItemProvider(HtmlFactsService htmlFactsService)
+        {
+            if (htmlFactsService is null)
+            {
+                throw new ArgumentNullException(nameof(htmlFactsService));
+            }
+
+            _htmlFactsService = htmlFactsService;
+        }
+
+        public override IReadOnlyList<RazorCompletionItem> GetCompletionItems(RazorSyntaxTree syntaxTree, TagHelperDocumentContext tagHelperDocumentContext, SourceSpan location)
+        {
+            var completions = new List<RazorCompletionItem>();
+            if (syntaxTree == null)
+            {
+                return completions;
+            }
+
+            var change = new SourceChange(location, string.Empty);
+            var owner = syntaxTree.Root.LocateOwner(change);
+
+            if (owner == null)
+            {
+                Debug.Fail("Owner should never be null.");
+                return completions;
+            }
+
+            if (AtTextCompletionPoint(owner))
+            {
+                var parent = owner.Parent;
+                if (_htmlFactsService.TryGetElementInfo(parent, out var containingTagNameToken, out var attributes) &&
+                    containingTagNameToken.Span.IntersectsWith(location.AbsoluteIndex))
+                {
+                    if (TextTag.StartsWith(containingTagNameToken.Content, StringComparison.OrdinalIgnoreCase)) {
+                        var completionDisplayText = TextTag;
+                        var completionItem = new RazorCompletionItem(
+                            completionDisplayText,
+                            completionDisplayText,
+                            RazorCompletionItemKind.Text,
+                            ElementCommitCharacters);
+                        var completionDescription = new DirectiveCompletionDescription("Create a inline HTML element without tags.");
+                        completionItem.SetDirectiveCompletionDescription(completionDescription);
+                        completions.Add(completionItem);
+                    }
+                }
+            }
+
+            return completions;
+        }
+
+        internal static bool AtTextCompletionPoint(Microsoft.AspNetCore.Razor.Language.Syntax.SyntaxNode owner)
```

Comment:
Why the fully qualified name? `Microsoft.AspNetCore.Razor.Language.Syntax.SyntaxNode`

Reply:
Otherwise it defaults to the namespace `Microsoft.CodeAnalysis.SyntaxNode` which isn't what we're looking for. Would you like me to change it to something else?

Reply:
Use usings alias like this https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/DefaultRazorFormattingService.cs#L17